_posts/blog/2017-08-014-upday_architecture.md

```diff
@@ -0,0 +1,106 @@
+---
+layout: post
+title: "Finding Yourself in Backend Architectural World"
+modified:
+categories: blog
+author: maria_fernandez_pajares
+excerpt: Our journey through the world of backend architecture in a few words.
+tags: [Architecture, Microservices, Backend]
+image:
+date: 2017-08-14T10:55:55+2
+---
+
+I have always wanted to write about our journey through the world of backend architectures. I guess you as developer have been facing problems related with this topic in your team and projects and
+are aware of lot of magical and trendy solutions, but sometimes those do not fit so well to your current situation. 
+
+This topic has not been -and it is not yet- an easy one for us neither, but after facing many problems we are finding ourselves in this architecture world.
+Therefore I would like to share a bit of our experiences and learnings from it in upday. 
+
+### Conway's Law's reversion
+
+In conferences I have attended I have heard references to what could be interpreted from [*Mel Conway's Law*](https://en.wikipedia.org/wiki/Conway%27s_law):
+
+>A backend architecture is a reflection of the team setup.
+
+and in my opinion I agree totally with it but not only that, **with time it may become the other way around:** 
+
+>the architecture can end up shaping the teams instead of the teams shaping the architecture.
+
+Let me dig a bit more in when we started to face this turn around and how we escaped from that:
+
+### Our firsts architectural steps
+
+In *upday* we started with a small team of developers tasked with creating a robust solution to fulfill our client’s requirements of reliability and providing good quality content to our readers. 
+As in most *startups*, everything started fast and in a rush. The solution we finally came up with consisted of two monolithic components written in *Java* and *Spring Boot*. 
+Both of them together being the core of our system. They were not the only ones. We also had other smaller components written in *Ruby* or *Java*, which took care of minor requirements. 
```

Comment:
Together they form the core of our system. But they were not the only ones. We also had other smaller components written in *Ruby* or *Java*, which took care of minor requirements.